assignments/this.js

```diff
@@ -1,26 +1,57 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. While in global scope the value of this will be the window.
+* 2. The object to the left of the dot in a function is this.
+* 3. A constructor function is a function template that returns new objects.
+* 4. Whenever .call or .apply is used in a function, this is explicitly defined.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
-
-// code example for Window Binding
+function Season(argument){
+    console.log(this);
+    return argument;
+} 
+Season('Winter');
 
 // Principle 2
-
-// code example for Implicit Binding
+const myObj = {
+    greeting: 'What\'s Up',
+    sayHello: function(name) {
+      console.log(`${this.greeting} my name is ${name}`);
+      console.log(this);
+    }
+}
+  myObj.sayHello('Ryan');
 
 // Principle 3
-
-// code example for New Binding
+function Homebody(does){
+    this.does = does,
```

Comment:
Great job with new binding. Remember that the `new` keyword will just set `this` to the new object that is created.